index/index.go

```diff
@@ -241,7 +242,32 @@ func isTextFile(filename string) (bool, error) {
 
 	buf = buf[:n]
 
-	return utf8.Valid(buf), nil
+	res, validSize := validUTF8(buf)
+	// if remaining non valid less than utf8.UTFMax probably we are read part of rune just assume it's valid
+	if n == filePeekSize && n-validSize < utf8.UTFMax {
+		res = true
+	}
+	return res, nil
+}
+
+// copy paste from go source https://golang.org/src/unicode/utf8/utf8.go?s=9677:9702#L387, with valid bytes size
+func validUTF8(p []byte) (bool, int) {
```

Comment:
Rather than rewriting `utf8.ValidUTF8`, is it possible to use `utf8.DecodeLastRune` to find the end of the last valid rune and then pass a slice of buf to `utf8.ValidUTF8`?